File: FitProj/app/src/main/java/com/teuskim/fitproj/common/CompatUtil.java
package com.teuskim.fitproj.common;

import android.os.Build;
import android.view.View;
import android.view.Window;

/**
 * 하위호환 필요한 메소드들 모음
 */
public class CompatUtil {

    public static void setElevation(View v, float elevation) {
        if (Build.VERSION.SDK_INT >= 21) {
            v.setElevation(elevation);
        }
    }

    public static void setStatusBarColor(Window window, int color) {
        if (Build.VERSION.SDK_INT >= 21) {
            window.setStatusBarColor(color);
        }
    }
}

File: FitProj_kotlin/app/src/main/java/com/teuskim/fitproj/common/CompatUtil.kt
package com.teuskim.fitproj.common

import android.os.Build
import android.view.View
import android.view.Window

/**
 * 하위호환 필요한 메소드들 모음
 */
object CompatUtil {

    fun setElevation(v: View, elevation: Float) {
        if (Build.VERSION.SDK_INT >= 21) {
            v.elevation = elevation
        }
    }

    fun setStatusBarColor(window: Window, color: Int) {
        if (Build.VERSION.SDK_INT >= 21) {
            window.statusBarColor = color
        }
    }
}
